From 0ddc7695789b83e955a899612a0a3fc9ad3742cf Mon Sep 17 00:00:00 2001
From: Konstantin Haase <konstantin.mailinglists@googlemail.com>
Date: Sun, 10 Mar 2013 13:08:47 +0100
Subject: [PATCH] avoid to_enum warning on 2.0

---
 test/response_test.rb | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/test/response_test.rb b/test/response_test.rb
index 32ab6ed..2c843d2 100644
--- a/test/response_test.rb
+++ b/test/response_test.rb
@@ -8,8 +8,7 @@ class ResponseTest < Test::Unit::TestCase
   end
 
   def assert_same_body(a, b)
-    enum = Enumerable.const_get(:Enumerator)
-    assert_equal enum.new(a).to_a, enum.new(b).to_a
+    assert_equal a.to_enum(:each).to_a, b.to_enum(:each).to_a
   end
 
   it "initializes with 200, text/html, and empty body" do
-- 
1.8.5.1

